paasta_tools/instance/kubernetes.py

```diff
@@ -424,6 +424,65 @@ def filter_actually_running_replicasets(
     ]
 
 
+def bounce_status(
+    service: str, instance: str, settings: Any,
+):
+    status: Dict[str, Any] = {}
+    job_config = kubernetes_tools.load_kubernetes_service_config(
+        service=service,
+        instance=instance,
+        cluster=settings.cluster,
+        soa_dir=settings.soa_dir,
+        load_deployments=True,
+    )
+    expected_instance_count = job_config.get_instances()
+    status["expected_instance_count"] = expected_instance_count
+    desired_state = job_config.get_desired_state()
+    status["desired_state"] = desired_state
+
+    kube_client = settings.kubernetes_client
+    if kube_client is None:
+        return status
```

Comment:
nit: Can we raise an exception (maybe `RuntimeError`?) here instead? This way, the caller gets the message that we can't access talk to k8s, instead of just getting an empty return value.

Reply:
I think the real way to do this is to control the HTTP response code -- let me see how easy that is

Reply:
Ok I raised a `RuntimeError` and then added some logic to catch it and reraise as an `ApiFailure`